pkg/apis/core/validation/validation.go

```diff
@@ -2711,6 +2719,71 @@ func ValidateVolumeDevices(devices []core.VolumeDevice, volmounts map[string]str
 	return allErrs
 }
 
+func validatePodResourceClaims(claims []core.PodResourceClaim, fldPath *field.Path) (sets.String, field.ErrorList) {
+	var allErrs field.ErrorList
+	podClaimNames := sets.NewString()
+	for i, claim := range claims {
+		allErrs = append(allErrs, validatePodResourceClaim(claim, podClaimNames, fldPath.Index(i))...)
+	}
+	return podClaimNames, allErrs
+}
+
+func validatePodResourceClaim(claim core.PodResourceClaim, podClaimNames sets.String, fldPath *field.Path) field.ErrorList {
```

Comment:
I know, and you know that `sets.String` is a `map` but we should still pass by pointer so it is clear thsat this func mutates it in the parent context,

Reply:
Changed.